tests/TestCase.php

```diff
@@ -2,9 +2,35 @@
 
 namespace Tests;
 
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+use Illuminate\Database\Eloquent\Relations\HasMany;
 use Illuminate\Foundation\Testing\TestCase as BaseTestCase;
 
 abstract class TestCase extends BaseTestCase
 {
     use CreatesApplication;
+
+    protected function test_hasMany_relation($related, $foreignKey, $relation)
+    {
+        $this->assertInstanceOf(HasMany::class, $relation);
+        $this->assertInstanceOf($related, $relation->getRelated());
+        $this->assertEquals($foreignKey, $relation->getForeignKeyName());
+    }
+
+    protected function test_belongsTo_relation($related, $foreignKey, $relation, $ownerKey)
+    {
+        $this->assertInstanceOf(BelongsTo::class, $relation);
+        $this->assertInstanceOf($related, $relation->getRelated());
+        $this->assertEquals($ownerKey, $relation->getOwnerKeyName());
+        $this->assertEquals($foreignKey, $relation->getForeignKeyName());
+    }
+
+    protected function test_belongsToMany_relation($related, $foreignKey, $relation, $relatedKey)
+    {
+        $this->assertInstanceOf(BelongsToMany::class, $relation);
+        $this->assertInstanceOf($related, $relation->getRelated());
+        $this->assertEquals($relatedKey, $relation->getRelatedPivotKeyName());
+        $this->assertEquals($foreignKey, $relation->getForeignPivotKeyName());
+    }
```

Comment:
giải thích các function này, em test những gì

Reply:
ở đây em test một `relation` `belongsToMany` và cụ thể test 4 thứ:
- Kiểm tra xem `relation` có phải instance của `belongsToMany`
- Kiểm tra xem tên một `model` cụ thể có phải instance của `model` được `related` với `relation`
- Kiểm tra giá trị khoá trong bảng trung gian mối quan hệ `many to many` đối với bảng còn lại trong `relation`
- Kiểm tra giá trị khoá trong bảng trung gian mối quan hệ `many to many` đối với bảng hiện tại đang test